packages/commons/src/auth/jwk.ts

```diff
@@ -0,0 +1,33 @@
+import crypto, { JsonWebKey } from "crypto";
+import {
+  jwkDecodingError,
+  notAllowedPrivateKeyException,
+} from "pagopa-interop-models";
+
+export const decodeBase64ToPem = (base64String: string): string => {
+  try {
+    const cleanedBase64 = base64String.trim();
+    const decodedBytes = Buffer.from(cleanedBase64, "base64");
+    return decodedBytes.toString("utf-8");
+  } catch (error) {
+    throw jwkDecodingError(error);
+  }
+};
+
+// This method is to check if the key is public and to create the JWK If you don't add the if condition, createPublicKey turns any key into a public key
+export const createJWK = (pemKey: string): JsonWebKey => {
+  if (
+    pemKey.includes(
+      "-----BEGIN RSA PUBLIC KEY-----" || "-----BEGIN PUBLIC KEY-----"
+    )
+  ) {
+    return crypto.createPublicKey(pemKey).export({ format: "jwk" });
+  } else {
+    throw notAllowedPrivateKeyException();
+  }
+};
+
+export const calculateKid = (jwk: JsonWebKey): string => {
```

Comment:
This function does not implement [RFC7638](https://www.rfc-editor.org/rfc/rfc7638.html)
To obtain a valid kid you should add the following step:
- sort jwk keys lexicographically
- encode the sha256 hash in Base64 URL-safe

I strongly recommend to add a specific test for this part using